src/search/tests/test_unified_search.py

```diff
@@ -410,3 +444,608 @@ def test_valid_search_requires_opensearch(self):
         # Just a placeholder to document that integration tests
         # should be run manually or in CI with OpenSearch
         pass
+
+
+class AuthorNameSearchTests(TestCase):
+    """Tests specifically for author name search functionality."""
+
+    def setUp(self):
+        self.query_builder = UnifiedSearchQueryBuilder()
+
+    def test_author_last_name_only_query(self):
+        """Test query for last name only (e.g., 'Smith')."""
+        query = self.query_builder.build_document_query("Smith")
+        qd = query.to_dict()
+
+        # Helper to recursively find all match and multi_match queries
+        def find_author_queries(node, results=None):
```

Comment:
Can we simplify our checks in such a way where we don't have to traverse complex dictionary structures?